### PR TITLE
Support BatchNorm with runtime (non-static) inputs

### LINKED ISSUE #20: ONNX error "BatchNorm: weight tensor must be present"
**Describe the bug**
When trying to convert onnx model to Burn, I'm encountering this panic:
```
  WARN burn_import::onnx::dim_inference: Must implement dimension inference for Slice
  WARN burn_import::onnx::dim_inference: Must implement dimension inference for Slice
  ERROR burn_import::logger: PANIC => panicked at burn-import-0.12.1\src\onnx\dim_inference.rs:311:9:
  Only tensor input is valid Argument

  --- stderr
  thread 'main' panicked at burn-import-0.12.1\src\onnx\dim_inference.rs:311:9:
  Only tensor input is valid Argument
```
The `node_input` on that line is `{ name: "cast2_out1", ty: Scalar(Int64), value: None, passed: true }`

The model I'm trying to convert is [gmflow-scale2-regrefine6-mixdata-train320x576-4e7b215d.onnx](https://drive.google.com/file/d/1raYbnf8nnxg9mb4W6bg66PfkB1woK8QB/view?usp=sharing)

I also have alternative onnx model exported by someone else, which errors with a different message, might also be worth looking into: [gmflow-scale1-mixdata-train320x576-4c3a6e9a_1x3x480x640_sim.onnx](https://drive.google.com/file/d/1ARg0gCvOouXyHEuUgDxCJ_lRbbezbl6A/view?usp=sharing)

**To Reproduce**
1. Create a build.rs with:
```rust
fn main() {
    ModelGen::new()
        .input("gmflow-scale2-regrefine6-mixdata-train320x576-4e7b215d.onnx")
        .out_dir("model/")
        .run_from_script();
}
```
2. Watch it fail

**Expected behavior**
I expected it to finish conversion successfully

**Desktop (please complete the following information):**
 - OS: Windows 11, Rust 1.77.1, Burn 0.12

**Additional context**
Model exported using `torch.onnx.export` from [this repo](https://github.com/autonomousvision/unimatch/)

### PR DESCRIPTION
**Pull request by antimora**
## Summary
- Fixes `extract_config` panic when BatchNorm's scale/bias/mean/var inputs come from other nodes (runtime) rather than static initializers
- Changes `BatchNormConfig` to an enum with `Static` and `Runtime` variants
- Static path (unchanged): generates `BatchNorm<B>` module
- Runtime path (new): generates inline batch normalization math in `forward()`
- Adds integration test with runtime BatchNorm inputs verified against ONNX ReferenceEvaluator

## Test plan
- [x] `cargo test -p onnx-ir` — existing + new unit tests pass
- [x] `cargo test -p burn-onnx` — existing + new snapshot tests pass
- [x] `cargo test -p onnx-tests` — existing batch_norm + new batch_norm_runtime tests pass
- [x] Verified gmflow model no longer panics on BatchNorm parsing

Fixes #20

### CODE REVIEW

File: crates/onnx-ir/src/node/batch_norm.rs [hunk 1]
```diff
@@ -127,8 +143,20 @@ impl NodeProcessor for BatchNormProcessor {
...
+                num_features,
+                epsilon as f64,
+                momentum as f64,
+            )))
+        } else {
+            Ok(BatchNormConfig::Runtime(BatchNormRuntimeConfig::new(
+                epsilon as f64,
+                momentum as f64,
+            )))
+        }
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/batch_norm.rs]: > The logic assumes that if scale (input[1]) is static, then all weight inputs are static...

Good catch. Fixed in 1d3e86a — `extract_config` now checks all four weight inputs (1–4), and `lift_constants` only lifts when all four are constants to avoid partial lifting.

File: crates/burn-onnx/src/burn/node/batch_norm.rs [hunk 2]
```diff
@@ -131,4 +205,54 @@ mod tests {
...
+                let scale = scale.clone().unsqueeze_dims(&[0isize, 2isize, 3isize]);
+                let bias = bias.clone().unsqueeze_dims(&[0isize, 2isize, 3isize]);
+                let mean = mean.clone().unsqueeze_dims(&[0isize, 2isize, 3isize]);
+                let var = var.clone().unsqueeze_dims(&[0isize, 2isize, 3isize]);
+                (input.clone() - mean) / (var + 0.00001f64).sqrt() * scale + bias
+            };
+            output
+        }
+        ");
+    }
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-onnx/src/burn/node/batch_norm.rs:307]: Added snapshot tests for rank-3 and rank-5 in 1d3e86a, verifying the `unsqueeze_dims` logic generates the correct dimensions for each rank.

File: crates/burn-onnx/src/burn/node/batch_norm.rs [hunk 3]
```diff
@@ -11,81 +12,139 @@ impl NodeCodegen for onnx_ir::node::batch_norm::BatchNormalizationNode {
...
+                let rank = match &self.inputs[0].ty {
+                    ArgType::Tensor(t) => t.rank,
+                    _ => panic!("BatchNorm input must be a tensor"),
+                };
+
+                // Build the reshape dimensions: [1, C, 1, 1, ...] for broadcasting
+                // scale/bias/mean/var are 1D tensors of shape [C], need to broadcast
+                // to match input shape [N, C, D1, D2, ...]
+                let unsqueeze_dims: Vec<isize> = {
+                    // For rank-D input, unsqueeze dim 0 and dims 2..rank-1
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-onnx/src/burn/node/batch_norm.rs:69]: The Rust range `2..rank` on the next line makes this unambiguous. Not worth a comment change.
